src/queries/get-users.ts

```diff
@@ -1,22 +1,45 @@
 import { ContextReturn } from '../apollo-context/context-return';
 import { AppDataSource } from '../data-source';
 import { User } from '../entity/User';
+import { BadRequestError } from '../errors/bad-request.error';
 import { errorMessages } from '../errors/error-messages';
 import { UnauthorizedError } from '../errors/unauthorized.error';
+import { UsersPage } from './users-page-type';
 
-export async function getUsers(parent: any, args: { limit?: number }, context: ContextReturn) {
+export async function getUsers(parent: any, args: { limit?: number; index?: number }, context: ContextReturn) {
   if (!context.userId) {
     throw new UnauthorizedError(errorMessages.notAuthenticated);
   }
 
   const defaultLimit = 5;
+  const limit = args.limit ? args.limit : defaultLimit;
+  const index = args.index ? args.index : 1;
+
+  const usersCount = await AppDataSource.manager.getRepository(User).count();
+  const maxIndex = Math.ceil(usersCount / limit);
+
+  if (index > maxIndex) {
+    throw new BadRequestError(errorMessages.indexInvalid);
+  }
+
+  const skip = (index - 1) * limit;
+
   const usersList = await AppDataSource.manager.getRepository(User).find({
-    skip: 0,
-    take: args.limit ? args.limit : defaultLimit,
+    skip,
+    take: limit,
     order: {
-      name: 'DESC',
+      name: 'ASC',
     },
   });
 
-  return usersList;
+  const after = usersCount - skip - limit;
+
+  const response = {
+    total: usersCount,
+    before: skip,
+    after: after < 0 ? 0 : after,
+    users: usersList,
+  } as UsersPage;
```

Comment:
```suggestion
  const response: UsersPage = {
    total: usersCount,
    before: skip,
    after: after < 0 ? 0 : after,
    users: usersList,
  };
```

Cuidado com os casts, a maioria deles são evitáveis.

Reply:
Corrigido.